sklearn/utils/estimator_checks.py

```diff
@@ -265,7 +265,31 @@ def _yield_all_checks(name, estimator):
     yield check_fit_idempotent
 
 
-def check_estimator(Estimator):
+def readable_check_estimator_ids(val):
+    """Create readable pytest ids for `check_estimator` when
```

Comment:
It's also used internally without reference to ``check_estimator``.

How about

```
"""Create pytest ids for checks.

Returns string representations for pytest tests, to be used as id.
Use together with ``check_estimator(..., generate_only=True)``
"""
```

Though that makes documenting ``val`` possibly more difficult. But on the other hand you could just directly document what ``val`` needs to be instead of defining it implicitly.

Also: missing ``Returns`` section.